src/utils/request.ts

```diff
@@ -1,11 +1,19 @@
+export class ResponseError extends Error {
```

Comment:
Any thoughts on using [`axios`](https://github.com/axios/axios) instead of a custom request util? Seems to be a JS/TS industry standard practice at this point.

Reply:
Yeah but since it was already like this in the original boilerplate I didn't wanna change. axios would have been much simpler.